homework/index.js

```diff
@@ -29,19 +29,124 @@
+      createAndAppend('td', parent, {
+        text: `${textKey}: `,
+        class: 'label',
+      });
+      createAndAppend('td', parent, {
+        text: textContent,
+      });
+    }
+    createTd('Description', data[index].description);
+    createTd('Forks', data[index].forks);
+    createTd('Updated', data[index].updated_at);
+  }
+
+  // contributors list
+  function listContributors(index, data, container) {
+    const contributorUrl = data[index].contributors_url;
+    fetchJSON(contributorUrl, (err, contributors) => {
+      const root = document.getElementById('root');
+      if (err) {
+        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+      } else {
+        const contributorsData = createAndAppend('div', container, {
+          class: 'right-div white-frame',
+        });
+        createAndAppend('p', contributorsData, {
+          text: 'contributions',
+          class: 'contributor-header',
+        });
+        const ul = createAndAppend('ul', contributorsData, {
+          class: 'contributor-list',
+        });
+        contributors.forEach(contributor => {
+          const li = createAndAppend('li', ul, {
+            class: 'contributor-item',
+          });
+          // The open() method creates a new secondary browser window,
+          // a new blank, empty window (URL about:blank) is created with the default toolbars of the main window.
+          li.addEventListener('click', () => {
+            window.open(contributor.html_url);
+          });
+
+          createAndAppend('img', li, {
+            src: contributor.avatar_url,
+            alt: "contributor's profile",
+            class: 'contributor-avatar',
+          });
+          const liDiv = createAndAppend('div', li, { class: 'contributor-data' });
+          createAndAppend('div', liDiv, { text: contributor.login });
+          createAndAppend('div', liDiv, {
+            text: contributor.contributions,
+            class: 'contributor-badge',
+          });
+        });
+      }
+    });
+  }
+
+  // create the select options and the event listener for changing repositories
+  function selectAndChange(data, container, header) {
+    const select = createAndAppend('select', header, { class: 'repository-selector' });
+    // localeCompare enables a case-insensitive sort of an array.
+    const repositories = data.sort((a, b) => a.name.localeCompare(b.name));
+    repositories.forEach((repository, index) => {
+      createAndAppend('option', select, { text: repository.name, value: index });
+    });
+    createDetailsTable(0, data, container);
+    listContributors(0, data, container);
+
+    select.addEventListener('change', event => {
+      while (container.firstChild) {
+        container.removeChild(container.firstChild);
+      }
+      // event.target refers to the clicked select repository
+      // It will fire when each select repository is clicked
+      const index = event.target.value;
```

Comment:
Alternatively, you can also get this value with `select.value`. But your method is fine too.

Reply:
Much more easier, thanks